code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -630,119 +557,112 @@ GLOBAL_VAR_INIT(dynamic_forced_storyteller, null)
-			log_threat("[new_rule.ruletype] - <b>[new_rule.name]</b> [new_rule.cost] threat", verbose = TRUE)
+			spend_midround_budget(new_rule.cost)
+			threat_log += "[worldtime2text()]: Forced rule [new_rule.name] spent [new_rule.cost]"
+			new_rule.pre_execute(population)
 			if (new_rule.execute()) // This should never fail since ready() returned 1
-				if(new_rule.flags & HIGHLANDER_RULESET)
-					highlander_executed = TRUE
+				if(new_rule.flags & HIGH_IMPACT_RULESET)
+					high_impact_ruleset_executed = TRUE
 				else if(new_rule.flags & ONLY_RULESET)
 					only_ruleset_executed = TRUE
 				log_game("DYNAMIC: Making a call to a specific ruleset...[new_rule.name]!")
-				SSblackbox.record_feedback("associative","dynamic_rulesets",1,new_rule.get_blackbox_info())
 				executed_rules += new_rule
-				current_rules += new_rule
+				if (new_rule.persistent)
+					current_rules += new_rule
 				return TRUE
 		else if (forced)
-			log_game("DYNAMIC: The ruleset [new_rule.name] couldn't be executed due to lack of eligible players.")
-	return FALSE
-
-/// Mainly here to facilitate delayed rulesets. All midround/latejoin rulesets are executed with a timered callback to this proc.
-/datum/game_mode/dynamic/proc/execute_midround_latejoin_rule(sent_rule)
-	var/datum/dynamic_ruleset/rule = sent_rule
-	if (rule.execute())
-		message_admins("DYNAMIC: Injected a [rule.ruletype == "latejoin" ? "latejoin" : "midround"] ruleset [rule.name].")
-		log_game("DYNAMIC: Injected a [rule.ruletype == "latejoin" ? "latejoin" : "midround"] ruleset [rule.name].")
-		log_threat("[rule.ruletype] [rule.name] added [rule.cost]", verbose = TRUE)
-		if(rule.flags & HIGHLANDER_RULESET)
-			highlander_executed = TRUE
-		else if(rule.flags & ONLY_RULESET)
-			only_ruleset_executed = TRUE
-		if(rule.ruletype == "Latejoin")
-			var/mob/M = pick(rule.candidates)
-			message_admins("[key_name(M)] joined the station, and was selected by the [rule.name] ruleset.")
-			log_game("DYNAMIC: [key_name(M)] joined the station, and was selected by the [rule.name] ruleset.")
-		executed_rules += rule
-		SSblackbox.record_feedback("associative","dynamic_rulesets",1,rule.get_blackbox_info())
-		rule.candidates.Cut()
-		if (rule.persistent)
-			current_rules += rule
-		return TRUE
-	stack_trace("The [rule.ruletype] rule \"[rule.name]\" failed to execute.")
+			log_game("DYNAMIC: The ruleset [new_rule.name] couldn't be executed due to lack of elligible players.")
 	return FALSE
 
 /datum/game_mode/dynamic/process()
-	if (pop_last_updated < world.time - (120 SECONDS))
-		pop_last_updated = world.time
-		update_playercounts()
-
 	for (var/datum/dynamic_ruleset/rule in current_rules)
 		if(rule.rule_process() == RULESET_STOP_PROCESSING) // If rule_process() returns 1 (RULESET_STOP_PROCESSING), stop processing.
 			current_rules -= rule
-			SSblackbox.record_feedback("tally","dynamic",1,"Rulesets finished")
-			SSblackbox.record_feedback("associative","dynamic_rulesets_finished",1,rule.get_blackbox_info())
```

Comment:
for sure add the first one, but ditto with get_blackbox_info

Reply:
Added the first, kept second removed